README.md

```diff
@@ -100,6 +100,101 @@ You are free to implement the genome classes however you want, and using whateer
+
+#### active_te
+
+active_te has a complexity of O(p)
+
+Getting the key list from the dictionary, and returning the result both run in O(1) complexity. We assume that we have to run through the key dict_list to get it into a list, in which case it wil have O(p) complexity.
+
+#### len
+
+len runs in O(1), because evaluating the length of a list has O(1) complexity.
+
+#### str
+
+str runs in O(n)
+
+str uses the join function, which has to run through each element in the genome to make it into a string. Therefore we expect the function ro run in O(n).
+
+### Doubly-Linked-List Genome
+
+#### init
+
+init has a complexity of O(n).
+
+Creating the genome by running through each element and appending takes n * O(1) = O(n) complexity. Creating hte ID and the TE-dictionary takes O(1).
+
+#### insert_te
+
+insert_te has a complecity of O(n)
+
+We run through each element in the key-list (O(p)), checking if the new te either overlaps, or pushes the existing te's. This runs in max O(p). If there is a overlap, we run the disable_te function, that run in O(n). Creating the new TE takes O(m). Finding the start, where we want to insert our new TE takes a maximal O(n) complexity, if start is at the end of the genome. Inserting the new TE takes O(1). All of these run-times can then be reduced to O(n).
+
+#### copy_te
+
+copy_te has a complexity of O(n).
+
+checking the keys has a maximum run-time of O(p), but our worst-case run-time must call the insert_te function. Therefore the run time of copy_te is defined by insert_te, which is defined by disable_te. Disable_te has a maximum complexity of O(n), and therefore, so does copy_te. 
+
+#### disable_te
+
+disable_te has a complexity of O(n).
+
+Checking the key values takes a maximum of O(p) complexity. Only one key will fit our TE ID, so creating it within the loop only takes O(m) complexity. We create a **while i < start** loop, where i starts at 0, and has a maximum complexity of O(n), when the TE starts at the end of the genome. The **while j < length** loop runs in O(m) time. inserting the new disabled te should run in O(1), as should removing the now inactive key from the dictionary. The worst-time complexity for disable, is therefore proportional to the size of hte genome. 
+
+
+#### active_te
+
+active_te has a complexity of O(p)
+
+We are still using a dictionary to present our active TE's, so the run time and reasoning is the same as before.
+
+#### len
+
+len has a complexity of O(n).
+
+To get the length, we must loop through each element in the genome, creating a complecity of O(n).
+
+#### str
+
+str has a complexity of O(n^2)
+
+Because we loop through the DLList, and use the += operation for the values, we end up with a complexity of O(n^2). This is not optimal, and might be optimised by first creating a list of the elements (O(n)), and then using the join function (O(1)).
```

Comment:
Actually, `+=` on strings has used the dynamic array optimisation for the last couple of Python version, so it does run $O(n)$.